tour_model_eval/data_preprocessing.py

```diff
@@ -0,0 +1,54 @@
+import emission.storage.decorations.analysis_timeseries_queries as esda
+import emission.analysis.modelling.tour_model.cluster_pipeline as pipeline
+import emission.analysis.modelling.tour_model.similarity as similarity
+import pandas as pd
+from sklearn.model_selection import KFold
+
+
+# - trips: all trips read from database
+# - filter_trips: valid trips that have user labels and are not points
+def filter_data(user,radius):
+    trips = pipeline.read_data(uuid=user, key=esda.CONFIRMED_TRIP_KEY)
+    non_empty_trips = [t for t in trips if t["data"]["user_input"] != {}]
+    non_empty_trips_df = pd.DataFrame(t["data"]["user_input"] for t in non_empty_trips)
+    valid_trips_df = non_empty_trips_df.dropna(axis=0, how='any', thresh=None, subset=None, inplace=False)
+    valid_trips_idx_ls = valid_trips_df.index.tolist()
+    valid_trips = [non_empty_trips[i]for i in valid_trips_idx_ls]
+
+    # similarity codes can filter out trips that are points in valid_trips
+    sim = similarity.similarity(valid_trips, radius)
+    filter_trips = sim.data
```

Comment:
I gave this feedback earlier, but it is better to split `similarity.similarity` to have the filtering and binning code separately, so you can call only the filtering code here.

That is both more clear and more performant.

Reply:
Move the current filtering implementation in similarity into a separate function called `filter_distance`.
- Call `filter_distance` from your code
- Change the `similarity.similarity` first step to call `filter_distance`

Reply:
Move the current filtering implementation in similarity into a separate function called `filter_distance`.
- Call `filter_distance` from your code
- Change the `similarity.similarity` first step to call `filter_distance`